## tl;dr

EJX — реальный крупный конкурент, но его 1,37 млн MAU нельзя приписать офферу 100 ₽: одновременно шли Telegram Ads, миграция Dessly и дорогие розыгрыши. Копировать стоит простоту интерфейса и фиксированную награду. Рекомендация DoodleVPN: одна реферальная Mini App-страница и 30-дневный тест **«первая оплата друга → 100 ₽ на следующее платное продление»** поверх текущих 20%, без бесплатных дней.

## Context & Methods

### Key Assumptions

- DoodleVPN: 57 ordinary first-paid friends in July and 6 referrers at the two-friend threshold, from the validated July referral analysis.
- Current Doodle referral spending converts balance at the discounted annual daily rate: `22.99 × 70% / 365` USD/day.
- EJX public offer is 100 ₽ on internal balance and the visible device tariff is 15 ₽/day. Public evidence does not prove cash withdrawal.
- External scale observations are descriptive, not causal attribution. Several growth levers changed simultaneously.

## Data

Public competitor facts and Doodle unit-economics assumptions used in the decision.

In [1]:
competitor_facts = [{"fact": "Telegram Mini App MAU", "value": "1 368 742", "evidence": "Публичная Telegram-страница Mini App, проверено 31 июля 2026", "implication": "Масштаб реален, но MAU не равно платящим"}, {"fact": "Telegram Ads", "value": "21 креатив / 156 каналов", "evidence": "tgads.net, кампания с мая 2026", "implication": "Рост не был безбюджетным"}, {"fact": "Рост новостного канала", "value": "+575 308 за 30 дней", "evidence": "Telemetr, срез 30–31 июля 2026", "implication": "Был мощный внешний приток, а не только рефералы"}, {"fact": "Миграция старой базы", "value": "Dessly → EJX", "evidence": "Публичные посты DesslyVPN", "implication": "Часть MAU перенесена из существующего продукта"}, {"fact": "Реферальный розыгрыш", "value": "850 000 ₽ + техника", "evidence": "Публичный анонс EJX", "implication": "Виральность субсидировалась отдельным бюджетом"}, {"fact": "Публичная награда", "value": "100 ₽ на баланс", "evidence": "Пост EJX и экран Mini App", "implication": "Простая и мгновенно понятная единица ценности"}]
for row in competitor_facts:
    print(f"{row['fact']}: {row['value']} — {row['implication']}")

Telegram Mini App MAU: 1 368 742 — Масштаб реален, но MAU не равно платящим
Telegram Ads: 21 креатив / 156 каналов — Рост не был безбюджетным
Рост новостного канала: +575 308 за 30 дней — Был мощный внешний приток, а не только рефералы
Миграция старой базы: Dessly → EJX — Часть MAU перенесена из существующего продукта
Реферальный розыгрыш: 850 000 ₽ + техника — Виральность субсидировалась отдельным бюджетом
Публичная награда: 100 ₽ на баланс — Простая и мгновенно понятная единица ценности


## Results

The same nominal 100 ₽ purchases very different service value in the two products.

In [2]:
usd_rub = 78.4049
ejx_days = 6.666666666666667
doodle_days = 28.927616276129395
reward_exposure = [{"priority": 1, "model": "100 ₽ в текущий Doodle-баланс", "service_months": 55.0, "free_days_per_friend": 28.9, "july_paid_friends": 57, "cash_discount_rub": 0, "decision": "Не запускать"}, {"priority": 2, "model": "30 дней за каждого", "service_months": 57.0, "free_days_per_friend": 30.0, "july_paid_friends": 57, "cash_discount_rub": 0, "decision": "Не запускать"}, {"priority": 3, "model": "14 дней за каждого", "service_months": 26.6, "free_days_per_friend": 14.0, "july_paid_friends": 57, "cash_discount_rub": 0, "decision": "Резерв, но не первый тест"}, {"priority": 4, "model": "2 друга → 30 дней", "service_months": 6.0, "free_days_per_friend": 15.0, "july_paid_friends": 57, "cash_discount_rub": 0, "decision": "Fallback"}, {"priority": 5, "model": "100 ₽ на следующее продление", "service_months": 0.0, "free_days_per_friend": 0.0, "july_paid_friends": 57, "cash_discount_rub": 5700, "decision": "Рекомендация"}]
print('EJX visible value, device-days:', round(ejx_days, 1))
print('Doodle current-wallet value, days:', round(doodle_days, 1))
for row in reward_exposure:
    print(row['model'], row['service_months'], 'service-months on July baseline')

EJX visible value, device-days: 6.7
Doodle current-wallet value, days: 28.9
100 ₽ в текущий Doodle-баланс 55.0 service-months on July baseline
30 дней за каждого 57.0 service-months on July baseline
14 дней за каждого 26.6 service-months on July baseline
2 друга → 30 дней 6.0 service-months on July baseline
100 ₽ на следующее продление 0.0 service-months on July baseline


In [3]:
max_budget_rub = 5000
usd_rub = 78.4049
contribution = 5.34
max_budget_usd = max_budget_rub / usd_rub
print('maximum budget USD:', round(max_budget_usd, 2))
print('conservative break-even incremental payers:', round(max_budget_usd / contribution, 1))

maximum budget USD: 63.77
conservative break-even incremental payers: 11.9


## Takeaways

- Literal 100 ₽ in the current Doodle balance equals about 28.9 free days, versus 6.7 visible device-days at EJX.
- On July's 57 paid-friend baseline, literal copying would create about 55 service-months before any incremental lift.
- A 100 ₽ renewal discount preserves a payment event and creates zero free service-months.
- With a 5,000 ₽ hard budget and $5.34 incremental contribution per new paid referral, conservative break-even is about 12 incremental payers at the absolute cap; with the expected 3,000–4,000 ₽ issued liability, it is about 8–10.
- The Mini App should initially be one referral page over the existing backend, not a bot migration or game.